observer/observer.go

```diff
@@ -0,0 +1,119 @@
+package observer
+
+import (
+	"context"
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cometbft/cometbft/libs/pubsub/query"
+	rpchttp "github.com/cometbft/cometbft/rpc/client/http"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	comettypes "github.com/cometbft/cometbft/types"
+)
+
+const ModuleNameObserver = "observer"
+
+type Observer struct {
+	logger        log.Logger
+	tmRpc         *rpchttp.HTTP
```

Comment:
nit: can we rename to cometRpc, trying to move new code away from old nomenclature